haystack/schema.py

```diff
@@ -1436,36 +1436,103 @@ def _build_document_metrics_df(
         if simulated_top_k_retriever != -1:
             documents = documents[documents["rank"] <= simulated_top_k_retriever]
 
+        # find out which label matched
+        def find_matched_label_idxs(row) -> List[int]:  # pylint: disable=too-many-return-statements
+            id_matches = [idx for idx, val in enumerate(row["gold_documents_id_match"]) if val == 1.0]
+            context_matches = [
+                idx for idx, val in enumerate(row["gold_contexts_similarity"]) if val > 65.0
+            ]  # TODO: hardcoded threshold for now, will be param of calculate_metrics
```

Comment:
Should this TODO be resolved now? Or should this be opened as a new issue?

Reply:
It's there for a future PR. With it this would get way too big